### DISCUSSION TITLE
Sync not possible anymore:  Microsoft OneDrive API returned an error

### DISCUSSION THREAD

**Discussion by eandy24** [abraunegg/onedrive#2159] [General]
Hi all,

I have set up onedrive several months ago and everything worked like a charme thanks to the very good manuals.  Really nice and helpful little app, thank you for that.

Since 3 weeks or so, it stopped working, and I have no idea why.

The error message is:
```
ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 403 (Forbidden)
  Error Reason:     Database Is Read Only
  Error Timestamp:  2022-09-28T13:00:21
  API Request ID:   8fcac371-0687-4275-853c-a31d664b1925
Uploading new file ./Bilder/.../20220612_183651000_iOS 2.jpg ...skipped
```
This message is then repeated continuously with other files...

What have I tried so far (several times)
- Completly removed onedrive and then did a clean install according to manual
- Reauth

I am starting "onedrive --synchronize --resync " as standard user

Config:
```
onedrive --display-config
Configuration file successfully loaded
onedrive version                             = v2.4.21-1+np1
Config path                                  = /home/eandy/.config/onedrive
Config file found in config path             = true
Config option 'sync_dir'                     = /mnt/WDE
Config option 'enable_logging'               = true
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     =
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 86400
Config option 'monitor_log_frequency'        = 5
Config option 'monitor_fullscan_frequency'   = 2
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = true
Config option 'check_nosync'                 = true
Config option 'check_nomount'                = true
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 3000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 777
Config option 'sync_file_permissions'        = 777
Config option 'space_reservation'            = 52428800
Config option 'application_id'               =
Config option 'azure_ad_endpoint'            =
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   =
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = true
sync_list contents:
#Synclist
/Bilder
/Homes
/Photos
!Downloads
!Music
!Storage
!Video
!Bilder
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false<
```
Ubuntu Version
```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
```
The system is fully patched and up-to-date

Sync Status
```
onedrive --display-sync-status
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Local directory is out of sync with OneDrive
Approximate data to download from OneDrive: 102557 KB
```
Has anyone any idea?

Thank you in advance!

Regards
Andreas

**Comment by abraunegg** [comment]
@eandy24 

> The error message is:
> 
> ```
> ERROR: Microsoft OneDrive API returned an error with the following message:
>   Error Message:    HTTP request returned status code 403 (Forbidden)
>   Error Reason:     Database Is Read Only
>   Error Timestamp:  2022-09-28T13:00:21
>   API Request ID:   8fcac371-0687-4275-853c-a31d664b1925
> Uploading new file ./Bilder/.../20220612_183651000_iOS 2.jpg ...skipped
> ```
> 
> This message is then repeated continuously with other files...

This is a OneDrive API problem ... because the 403 error is generated by Microsoft, not this client. It appears that maybe your OneDrive account has been put into read-only mode?

What happens happens when you login to OneDrive using your credentials - can you access your data & upload files via the web browser interface?





> I am starting "onedrive --synchronize --resync " as standard user

Please *stop* using `--resync` unless the application asks you to use it ..... do not *use* resync each and every time you run the application.

![image](https://user-images.githubusercontent.com/4956234/192882815-3dfeff2f-0bf8-4915-9084-cbaa32c3f231.png)

**Reply by eandy24** [reply]
@abraunegg 
Thank you for looking into this.
You're absolutely right: My account is locked... Didn't realized this yet... problem solved.

And: I used --resync in this case only, because the app told me to do this, before I only used   --synchronize ...

**Reply by abraunegg** [reply]
No problem - glad you have this sorted now
